akvo/rsr/filters.py

```diff
@@ -98,26 +98,45 @@ def get_locations(location, locations):
 def location_choices(qs):
     """From a queryset get possible location filter choices"""
 
-    if qs.model is Project:
-        location_model = ProjectLocation
-    elif qs.model is ProjectUpdate:
+    country_ids = (
+        get_recipient_country_ids(qs) if qs.model is Project else
```

Comment:
I think it's alright that filtering by location doesn't show all the projects. For instance, this might happen with keywords or sector filters too -- not all projects may have an associated keyword/sector.

To keep the filter simple, I ignored the Project Locations, but I think we can append that to the list of recipient countries. If we are searching on the Project Locations, too, I think they should always be shown in the list of locations for a project in it's listing in the project directory.